cwatm/hydrological_modules/water_demand/water_demand.py

```diff
@@ -1843,31 +2286,39 @@ def dynamic(self):
+                self.var.wwtEffluentsGenerated_industry = self.var.returnflowNonIrr.copy() * divideValues(frac_industry, frac_domestic + frac_industry)
+                self.var.wwtEffluentsGenerated = self.var.wwtEffluentsGenerated_domestic + self.var.wwtEffluentsGenerated_industry
+            
+                self.var.wwtSewerCollection_domestic = np.where(self.var.wwtColArea > 0,
+                                                       np.minimum(self.var.wwtEffluentsGenerated_domestic,
+                                                                  self.var.wwtEffluentsGenerated_domestic), 0.)
+                self.var.wwtSewerCollection_industry = np.where(self.var.wwtColArea > 0,
+                                                       np.minimum(self.var.wwtEffluentsGenerated_industry,
+                                                                  self.var.wwtEffluentsGenerated_industry), 0.) 
 
-                self.var.returnflowNonIrr = np.maximum(self.var.returnflowNonIrr - self.var.wwtSewerCollection, 0.)
-                self.var.wwtSewerCollection = self.var.wwtSewerCollection + self.var.wwtUrbanLeakage
                 self.model.wastewater_module.dynamic()
+                
+                self.var.returnflowNonIrr = np.maximum(self.var.returnflowNonIrr - self.var.wwtSewerCollection, 0.)
+                self.var.wwtSewerCollectedBySoruce = self.var.wwtEffluentsGenerated - self.var.returnflowNonIrr
 
             # returnflow to river and to evapotranspiration
             if self.var.includeIndusDomesDemand:  # all demands are taken into account
-                if self.var.includeWastewater:
+                #if self.var.includeWastewater:
                     # add uncollected wastewater
-                    uncollectedWWT = self.var.wwtSewerCollection * self.var.cellArea - \
-                                     self.var.wwtExportedCollected - self.var.wwtSewerCollected  # M3
-                    self.var.returnflowNonIrr += uncollectedWWT / self.var.cellArea
+                #    uncollectedWWT = self.var.wwtSewerCollection * self.var.cellArea - \
+                #                     self.var.wwtExportedCollected - self.var.wwtSewerCollected  # M3
+                #    self.var.returnflowNonIrr += uncollectedWWT / self.var.cellArea
                 self.var.returnFlow = self.var.returnflowIrr + self.var.returnflowNonIrr
             else:  # only irrigation is considered
                 self.var.returnFlow = self.var.returnflowIrr
 
             # add wastewater discharge to river to returnFlow - so they are sent to routing
             if self.var.includeWastewater & self.var.includeIndusDomesDemand:
-                self.var.returnFlow += self.var.wwtOverflowOutM
+                self.var.returnFlow += self.var.act_irrWithdrawalSW_month > self.var.irrWithdrawalSW_max
```

Comment:
To look into

Reply:
I am not sure how this had happened. I have switched the lines. (canceling 2321). I am also not quite sure the meaning of :

                self.var.returnFlow += self.var.act_irrWithdrawalSW_month > self.var.irrWithdrawalSW_max

Reply:
It has no meaning. This line didn't come from elsewhere, did it? It would be best to delete it instead of commenting it out.

Reply:
This line didn't come from elsewhere, did it? - I am not sure. I will search in the history